File: src/physics/models/ballState.ts
import { IBallState } from "../../types/ballState";
import { Ball, Coordinate } from "../../types/basics";

export class BallState implements IBallState {
  readonly ball: Ball;
  position: Coordinate;
  velocity: Coordinate;
  movementRotation: number;
  sideRotation: number;
  pocketed: boolean;

  constructor(ball: Ball) {
    this.position = { x: 0, y: 0 };
    this.velocity = { x: 0, y: 0 };
    this.movementRotation = 0;
    this.sideRotation = 0;
    this.pocketed = false;
    this.ball = ball;
  }

  //   pocket() {
  //     this.position.x = 0;
  //     this.position.y = 0;
  //     this.velocity.x = 0;
  //     this.velocity.y = 0;
  //     this.movementRotation = 0;
  //     this.sideRotation = 0;
  //     this.pocketed = true;
  //   }
}

File: src/physics/dinamics/BallState.java
package physics.dinamics;

import physics.statics.Ball;
import java.awt.Point;

public class BallState {
    private Ball ball;
    Point position;
    Point velocity;
    Float movementRotation;
    Float sideRotation;
    Boolean pocketed;

    BallState(Ball ball) {
        this.ball = ball;
    }

    public Ball getBall() {
        return ball;
    }

    public Boolean isStoped() {
        return false;
    }
}
